packages/vite/src/node/server/send.ts

```diff
@@ -69,7 +70,11 @@ export function send(
       content = getCodeWithSourcemap(
         type,
         content.toString(),
-        ms.generateMap({ source: urlWithoutTimestamp, hires: 'boundary' }),
+        ms.generateMap({
```

Comment:
Unrelated, but a question: Since #13514, if the JS code has no sourcemap (`map` is falsy), we're still creating a fallback sourcemap here.

If there's no sourcemap, shouldn't we don't need to do anything?

Reply:
Unfortunately, to make dev tools happy, we'll always have to generate a sourcemap (https://github.com/vitejs/vite/pull/13514#issuecomment-1592431496).
Note that we are skipping sourcemap generation for [`{ "mappings": "" }` which indicates generating a sourcemap is meaningless](https://rollupjs.org/plugin-development/#source-code-transformations).

Reply:
Hmm, thats feels like unnecessary overhead if the user never debugs a file, but good to know that it's needed for cases like this.